src/orion/benchmark/task/profet/profet_task.py

```diff
@@ -0,0 +1,817 @@
+    return torch.cat((x, log_var * torch.ones_like(x)), dim=1)
+
+
+AppendLayer.forward = _patched_forward
+
+
+InputType = TypeVar("InputType")
+
+
+class ProfetTask(BaseTask, Generic[InputType]):
+    def __init__(
+        self,
+        benchmark: str,
+        max_trials: int = 100,
+        task_id: int = 0,
+        seed: int = 123,
+        input_dir: Union[Path, str] = None,
+        checkpoint_dir: Union[Path, str] = None,
+        train_config: MetaModelTrainingConfig = None,
+        device: Union[torch.device, str] = None,
+    ):
+        super().__init__(max_trials=max_trials)
+        self.benchmark = benchmark
+        self.task_id = task_id
+        self.seed = seed
+        self.input_dir = Path(input_dir if input_dir else "profet_data")
+        self.checkpoint_dir = Path(checkpoint_dir or self.input_dir / "checkpoints")
+        # The config for the training of the meta-model.
+        # NOTE: the train config is used to determine the hash of the task.
+        # TODO: Also save the train_config in the `configuration` of the task.
+        self.train_config = train_config or MetaModelTrainingConfig()
+
+        # The parameters that have an influence over the training of the meta-model are used to
+        # create the filename where the model will be saved.
+        task_hash_params = dict(
+            benchmark=self.benchmark,
+            task_id=self.task_id,
+            seed=self.seed,
+            **asdict(self.train_config),
+        )
+        logger.info(f"Task hash params: {task_hash_params}")
+        task_hash = compute_identity(**task_hash_params)
+
+        filename = f"{task_hash}.pkl"
+
+        self.checkpoint_file = self.checkpoint_dir / filename
+        logger.info(f"Checkpoint file for this task: {self.checkpoint_file}")
+
+        self.device: torch.device
+        if isinstance(device, torch.device):
+            self.device = device
+        else:
+            self.device = torch.device(
+                device or ("cuda" if torch.cuda.is_available() else "cpu")
+            )
+
+        # TODO: Save those in the configuration dict?
+        self._np_rng: Optional[np.random.RandomState] = None
+        self._torch_rng_state: Optional[Tensor] = None
+        with self.seed_randomness():
+            if os.path.exists(self.checkpoint_file):
+                logger.info(
+                    f"Model has already been trained: loading it from file {self.checkpoint_file}."
+                )
+                self.net, self.h = load_task_network(self.checkpoint_file)
+            else:
+                logger.info(
+                    f"Checkpoint file {self.checkpoint_file} doesn't exist: re-training the model."
+                )
+                logger.info(f"Task hash params: {task_hash_params}")
+
+                self.checkpoint_file.parent.mkdir(exist_ok=True, parents=True)
+                # Need to re-train the meta-model and sample this task.
+                self.net, self.h = get_task_network(
```

Comment:
`get_task_network` should take care of seeding itself properly. If torch seeding is missed it should be added to this function. The seed_randomness could be a generic contextmanager function instead of a method, and used in `get_task_network`.

Reply:
This isn't currently a problem though. In the interest of time can I leave this as-is?

Reply:
Fixed: Using the new context manager.

Reply:
Thanks, let's wait to clarify discussion [here](https://github.com/Epistimio/orion/pull/632#discussion_r707545327) before closing this conversation.

Reply:
Replied on the other thread.

Reply:
Thanks!